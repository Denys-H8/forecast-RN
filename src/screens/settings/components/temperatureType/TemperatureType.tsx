import React from 'react';
import { View } from 'react-native';

import constants from 'src/constants';
import { CustomText } from 'src/components';
import { styles } from './temperatureType.styles';

interface IProps {
  type: string;
  isActive: string;
}

const TemperatureType: React.FC<IProps> = ({ type, isActive }) => {
  const value = type === 'metric' ? 'C' : 'F';
  return (
    <View style={styles.degreesContainer}>
      {isActive === type ? (
        <View style={styles.degreesContainer}>
          <CustomText style={[styles.celsius, { color: constants.MAIN_COLOR }]}>0</CustomText>
          <CustomText style={[styles.number, { color: constants.MAIN_COLOR }]}>{value}</CustomText>
        </View>
      ) : (
        <View style={styles.degreesContainer}>
          <CustomText style={styles.celsius}>0</CustomText>
          <CustomText style={styles.number}>{value}</CustomText>
        </View>
      )}
    </View>
  );
};

export default TemperatureType;
