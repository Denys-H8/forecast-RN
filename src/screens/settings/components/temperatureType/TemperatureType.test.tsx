import React from 'react';
import { render } from '@testing-library/react-native';
import TemperatureType from './TemperatureType';

describe('TemperatureType ', () => {
  test('should render active metric TemperatureType', () => {
    const { toJSON, queryByText } = render(<TemperatureType type="metric" isActive="metric" />);

    expect(toJSON()).toMatchSnapshot();
    expect(queryByText('C')).not.toBeNull();
    expect(queryByText('F')).toBeNull();
  });
  test('should render not active imperial TemperatureType', () => {
    const { toJSON, queryByText } = render(<TemperatureType type="imperial" isActive="metric" />);

    expect(toJSON()).toMatchSnapshot();
    expect(queryByText('C')).toBeNull();
    expect(queryByText('F')).not.toBeNull();
  });
});
