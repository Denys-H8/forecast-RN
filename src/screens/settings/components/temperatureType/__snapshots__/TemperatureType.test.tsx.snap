// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureType  should render active metric TemperatureType 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Array [
            Object {
              "color": "#8B8EA5",
              "fontFamily": "WorkSans-SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            },
            Object {
              "color": "#fff",
            },
          ],
        ]
      }
    >
      0
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Array [
            Object {
              "color": "#8B8EA5",
              "fontFamily": "WorkSans-SemiBold",
              "fontSize": 28,
              "lineHeight": 30,
            },
            Object {
              "color": "#fff",
            },
          ],
        ]
      }
    >
      C
    </Text>
  </View>
</View>
`;

exports[`TemperatureType  should render not active imperial TemperatureType 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#8B8EA5",
            "fontFamily": "WorkSans-SemiBold",
            "fontSize": 16,
            "lineHeight": 18,
          },
        ]
      }
    >
      0
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#8B8EA5",
            "fontFamily": "WorkSans-SemiBold",
            "fontSize": 28,
            "lineHeight": 30,
          },
        ]
      }
    >
      F
    </Text>
  </View>
</View>
`;
